src/liboslcomp/artic.cpp

```diff
@@ -474,17 +472,94 @@ ArticTranspiler::transpile_typecast_expression(ASTtypecast_expression* node)
 void
 ArticTranspiler::transpile_type_constructor(ASTtype_constructor* node)
 {
-    NOT_IMPLEMENTED;
+    std::vector<ASTNode::ref> args = {};
+    ASTNode::ref arg_node          = node->args();
+    while (arg_node) {
+        args.push_back(arg_node);
+        arg_node = arg_node->next();
+    }
+    source.add_source(artic_string(node->typespec()), "{");
+    if (node->typespec().is_triple() && args.size() == 1) {
+        args.push_back(args[0]);
+        args.push_back(args[0]);
+    }
+
+    for (size_t i = 0; i < args.size(); ++i) {
+        source.add_source(get_arg_name(node->typespec(), static_cast<int>(i)), " = ");
+        dispatch_node(args[i]);
+        source.add_source(", ");
+
+    }
+    source.add_source("}");
 }
+
+
 void
 ArticTranspiler::transpile_function_call(ASTfunction_call* node)
 {
-    NOT_IMPLEMENTED;
+    if(node->is_struct_ctr()){
+        auto constructor = ASTtype_constructor(nullptr, node->typespec(), node->args().get());
+        transpile_type_constructor(&constructor);
+    } else {
+        std::vector<ASTNode::ref> args = {};
```

Comment:
why store those intermediate results ? can't you dispatch directly ?

Reply:
If I do that I get a pointer exception because of the weird Ref counting the OSL SDk is doing internally with all AST nodes, and there doesnt seem to be a way to opt out of it

Reply:
never mind, you cant take the address of temporary objects, and I dont want to implement another constructor just in order to get a pointer (so writing
`transpile_type_constructor(&ASTtype_constructor(node));`
doesn't compile)